src/commands/publish/preview/mod.rs

```diff
@@ -3,89 +3,93 @@ use std::process::Command;
 mod http_method;
 pub use http_method::HTTPMethod;
 
+use crate::commands::build;
+use crate::commands::watch_and_build;
 use crate::commands::publish;
 
-use serde::Deserialize;
+use serde::{Deserialize, Serialize};
 use uuid::Uuid;
 
-use crate::commands;
 use crate::http;
+use crate::install;
+use crate::settings::project::get_project_config;
 use crate::settings::project::{Project, ProjectType};
 use crate::terminal::message;
 
-#[derive(Debug, Deserialize)]
-struct Preview {
-    pub id: String,
-}
+use std::sync::mpsc::channel;
+use std::thread;
+use ws::WebSocket;
 
 pub fn preview(
     project: &Project,
     method: Result<HTTPMethod, failure::Error>,
     body: Option<String>,
+    livereload: bool,
 ) -> Result<(), failure::Error> {
-    let create_address = "https://cloudflareworkers.com/script";
-
-    let client = http::client();
-
-    let project_type = &project.project_type;
-
-    commands::build(&project)?;
-
-    let res = match project_type {
-        ProjectType::Rust => client
-            .post(create_address)
-            .multipart(publish::build_multipart_script()?)
-            .send(),
-        ProjectType::JavaScript => client
-            .post(create_address)
-            .body(publish::build_js_script()?)
-            .send(),
-        ProjectType::Webpack => client
-            .post(create_address)
-            .multipart(publish::build_webpack_form()?)
-            .send(),
-    };
-
-    let p: Preview = serde_json::from_str(&res?.text()?)?;
+    build(&project)?; //do the initial build
 
     let session = Uuid::new_v4().to_simple();
 
     let preview_host = "example.com";
-    let https = 1;
-    let script_id = &p.id;
+    let https = true;
+    let script_id = &upload_and_get_id()?;
 
     let preview_address = "https://00000000000000000000000000000000.cloudflareworkers.com";
     let cookie = format!(
         "__ew_fiddle_preview={}{}{}{}",
```

Comment:
It's clearly out of scope for review but I went and read the implementation of this cookie serverside and, yeesh.

Reply:
@ejcx Thanks for the review, wanted to know if there's an internal ticket you made for the serverside cookie implementation, and if it's blocking this feature.